site/pages/api/create-consequence-network.test.ts

```diff
@@ -299,4 +301,47 @@ describe("create-consequence-network API", () => {
             Location: `${CREATE_CONSEQUENCE_NETWORK_PATH}/${defaultDisruptionId}/${defaultConsequenceIndex}?${returnPath}`,
         });
     });
+
+    it("should redirect to /type-of-consequence when all required inputs are passed and add another consequence is true", async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: { ...defaultNetworkData },
+            query: { addAnotherConsequence: "true" },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        await createConsequenceNetwork(req, res);
+
+        expect(upsertConsequenceSpy).toHaveBeenCalledTimes(1);
+        expect(upsertConsequenceSpy).toHaveBeenCalledWith(
+            networkToUpsert,
+            DEFAULT_ORG_ID,
+            mockSession.isOrgStaff,
+            false,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${TYPE_OF_CONSEQUENCE_PAGE_PATH}/${defaultDisruptionId}/1`,
+        });
+    });
+    it("should redirect to /type-of-consequence when all required inputs are passed and add another consequence is true and a template", async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: { ...defaultNetworkData },
+            query: { addAnotherConsequence: "true", template: "true" },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        await createConsequenceNetwork(req, res);
+
+        expect(upsertConsequenceSpy).toHaveBeenCalledTimes(1);
+        expect(upsertConsequenceSpy).toHaveBeenCalledWith(
+            networkToUpsert,
+            DEFAULT_ORG_ID,
+            mockSession.isOrgStaff,
+            true,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${TYPE_OF_CONSEQUENCE_PAGE_PATH}/${defaultDisruptionId}/1?template=true`,
```

Comment:
There are cases where the api redirects with return path query param - could we add these test cases?

Some cases are:
- Creating new disruption and then clicking change button from review disruption should contain return param with review page value
- Clicking change link from disruption details page should contain return param with disruption details page
- Clicking change link from review template page should include both template and return params
- Clicking change link for new disruptions from template page should include return params with disruption details page only

Reply:
as discussed